File: src/dotnet/Common/Models/Orchestration/EndpointSettings.cs
using FoundationaLLM.Common.Constants.Agents;

namespace FoundationaLLM.Common.Models.Orchestration
{
    /// <summary>
    /// Settings for an orchestration endpoint.
    /// </summary>
    public class EndpointSettings
    {
        /// <summary>
        /// Uri of the orchestration endpoint.
        /// </summary>
        public string? Endpoint { get; set; }

        /// <summary>
        /// API key for authorizing against an endpoint.
        /// </summary>
        public string? APIKey { get; set; }

        /// <summary>
        /// API version to use when accessing the endpoint.
        /// </summary>
        public string? APIVersion { get; set; }

        /// <summary>
        /// The type of authentication to use against the endpoint.
        /// This value should be either key or token.
        /// </summary>
        public string? AuthenticationType { get; set; }

        /// <summary>
        /// Type of operation the endpoint is performing.
        /// This value should be completions or chat.
        /// Default value is chat.
        /// </summary>
        public string OperationType { get; set; } = OperationTypes.Chat;
    }
}

File: src/python/PythonSDK/foundationallm/models/orchestration/endpoint_settings.py
from pydantic import BaseModel
from typing import Optional

from .operation_types import OperationTypes
from foundationallm.models.authentication import AuthenticationTypes

class EndpointSettings(BaseModel):
    endpoint: str
    api_key: Optional[str] = None
    api_type: Optional[str] = None
    api_version: Optional[str] = None
    authentication_type: str = AuthenticationTypes.TOKEN
    operation_type: str = OperationTypes.CHAT
    provider: str
